frontend/src/metabase/dashboard/hoc/DashboardControls.tsx

```diff
@@ -0,0 +1,92 @@
+/* This contains some state for dashboard controls on both private and embedded dashboards.
+ * It should probably be in Redux?
+ *
+ * @deprecated HOCs are deprecated
+ */
+import type { Location } from "history";
+import type { ComponentType } from "react";
+
+import { useSyncURLSlug } from "metabase/dashboard/components/DashboardTabs/use-sync-url-slug";
+import {
+  useDashboardNav,
+  useDashboardUrlParams,
+  useRefreshDashboard,
+} from "metabase/dashboard/hoc/controls";
+import type { DashboardControlsPassedProps } from "metabase/dashboard/hoc/types";
+import type { DashboardId } from "metabase-types/api";
+
+export const DashboardControls = (
+  ComposedComponent: ComponentType<DashboardControlsPassedProps>,
+) => {
+  function DashboardControlsInner({
+    dashboardId,
+    location,
+    ...props
+  }: {
+    dashboardId: DashboardId;
+    location: Location;
+  }) {
+    const queryParams = location.query;
+
+    const { refreshDashboard } = useRefreshDashboard({
```

Comment:
I'm not sure where else to put this. I think it's fine here since we still need it to refresh the dashboard